packages/backend-function/src/layer_parser.ts

```diff
@@ -0,0 +1,60 @@
+import { AmplifyUserError } from '@aws-amplify/platform-core';
+
+/**
+ * Parses Lambda Layer ARNs for a function
+ */
+export class FunctionLayerArnParser {
+  private arnPattern = new RegExp(
+    'arn:[a-zA-Z0-9-]+:lambda:[a-zA-Z0-9-]+:\\d{12}:layer:[a-zA-Z0-9-_]+:[0-9]+'
+  );
+
+  /**
+   * Parse the layers for a function
+   * @param layers - Layers to be attached to the function
+   * @param functionName - Name of the function
+   * @returns Valid layers for the function
+   * @throws AmplifyUserError if the layer ARN is invalid
+   * @throws AmplifyUserError if the number of layers exceeds the limit
+   */
+  parseLayers(
+    layers: Record<string, string>,
+    functionName: string
+  ): Record<string, string> {
+    const validLayers: Record<string, string> = {};
+
+    const uniqueArns = new Set<string>(Object.values(layers));
+    this.validateLayerCount(uniqueArns);
+
+    for (const [key, arn] of Object.entries(layers)) {
+      if (!this.isValidLayerArn(arn)) {
+        throw new AmplifyUserError('InvalidLayerArnFormatError', {
+          message: `Invalid ARN format for layer: ${arn}`,
+          resolution: `Update the layer ARN with the expected format: arn:aws:lambda:<current-region>:<account-id>:layer:<layer-name>:<version> for function: ${functionName}`,
+        });
+      }
+      validLayers[key] = arn;
+    }
```

Comment:
We are validating on the count of uniqueArns not the number of entries in the map. But later we are not deduping it later on. So for example if I provide a map of
```
{
  key1: arn1,
  key2: arn1,
  key3: arn1,
  key4: arn1,
  key5: arn1,
  key6: arn1,
}
```
our validation will succeed, but we will be passing/creating 6 reference layers to the `NodeJSFunction` constructor which may fail later.

Ideally we should dedupe here on the ARN and provide any of the key here since the key is only used for generating the name of the layer.

Reply:
oops thanks for the catch, updated this and also added a test to ensure unique Arn's are passed.